libyang/schema.py

```diff
@@ -411,6 +411,14 @@ def name(self) -> str:
     def module(self) -> Module:
         return self._module_from_parsed()
 
+    def parent_node(self) -> Optional["PNode"]:
```

Comment:
Is this related to the extension plugin?

Reply:
Yes, but indirectly. In real life, when you define your own extension plugin, on 90% of cases you would like to perform some validation whether extension is used correctly (e.g. is shall be used only on LEAF nodes etc..). This function is general way how to access the parent PNode to perform this.

Reply:
Then let's add a separate commit before this one adding support for parent_node for ExtensionParsed and ExtensionCompiled.

Reply:
Done